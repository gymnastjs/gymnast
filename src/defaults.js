// @flow
import { type ConfigContextType } from './types'

const defaults: {|
  ...ConfigContextType,
  gutter: number,
  fallbackDisplayKey: string,
  verticalGutter: number,
|} = {
  base: 8, // multiplier (in pixels) that all other size units use
  columns: 12, // number of columns used in the layout
  displayAliases: {
    // aliases used for the different display breakpoints:
    small: [
      {
        //   - "small" alias used when width is less than 600px
        maxWidth: '599px',
      },
      {
        maxDeviceWidth: '599px',
      },
    ],
    medium: [
      {
        //   - "medium" alias used when width is between 600px and 900px
        minWidth: '600px',
        maxWidth: '899px',
      },
      {
        minDeviceWidth: '600px',
        maxDeviceWidth: '899px',
      },
    ],
    large: [
      {
        //   - "large" alias used when width is equal or greater than 900px
        minWidth: '900px',
      },
      {
        minDeviceWidth: '900px',
      },
    ],
  },
  fallbackDisplayKey: 'default', // key to use when a display alias is omitted or non matching
  gutter: 3, // value (in base units) that separates columns horizontally
  spacingAliases: {
    // aliases used to indicate spacing values (margin/padding) in base
    XS: 0.5, // units.
    'S/2': 0.5,
    S: 1,
    'M/2': 1,
    M: 2,
    'L/2': 1.5,
    L: 3,
    'XL/2': 2,
    XL: 4,
    '2XL/2': 3,
    '2XL': 6,
  },
  verticalGutter: 3, // value (in base units) that separates columns vertically
}

export default defaults
