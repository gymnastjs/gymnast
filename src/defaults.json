{
  "spacingNames": {
    "XS": "4px",
    "S/2": "4px",
    "S": "8px",
    "M/2": "8px",
    "M": "16px",
    "L/2": "12px",
    "L": "24px",
    "XL/2": "16px",
    "XL":"32px",
    "XXL/2": "24px",
    "XXL":"48px"
  }
}
