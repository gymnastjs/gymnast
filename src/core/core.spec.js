import getCoreStyles from './index'
import defaults from '../defaults'

describe('getCoreStyles', () => {
  it('should return styles', () => {
    const { style } = getCoreStyles({ margin: 2 }, defaults)

    expect(style).toMatchSnapshot()
  })

  it('should return remaining props', () => {
    const { style, ...rest } = getCoreStyles(
      { margin: 2, foo: 'bar' },
      defaults
    )
    expect(style).toMatchSnapshot()
    expect(rest).toEqual({ foo: 'bar' })
  })

  it('should allow overriding base', () => {
    const { style } = getCoreStyles({ margin: 2, base: 10 }, defaults)

    expect(style).toMatchSnapshot()
  })
})
