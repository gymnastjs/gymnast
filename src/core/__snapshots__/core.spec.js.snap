// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCoreStyles should allow overriding base 1`] = `
Object {
  "borderBottomWidth": 20,
  "borderLeftWidth": 20,
  "borderRightWidth": 20,
  "borderTopWidth": 20,
}
`;

exports[`getCoreStyles should return remaining props 1`] = `
Object {
  "borderBottomWidth": 16,
  "borderLeftWidth": 16,
  "borderRightWidth": 16,
  "borderTopWidth": 16,
}
`;

exports[`getCoreStyles should return styles 1`] = `
Object {
  "borderBottomWidth": 16,
  "borderLeftWidth": 16,
  "borderRightWidth": 16,
  "borderTopWidth": 16,
}
`;
